src/plugins/ImportKeras/ImportKeras.js

```diff
@@ -0,0 +1,307 @@
+        this.core.setAttribute(this.importedArch, 'name', uniqueName);
+
+        this.logger.debug(`Added ${uniqueName} as a new architecture.`);
+        return Promise.resolve();
+    };
+
+    // This should add layers. constraints, initializers, regularizers as well as activations to the layer.
+    ImportKeras.prototype.addLayers = function () {
+        let layers = this.modelInfo.layers;
+        let layerToCreate = null;
+        this.layerInfo = {};
+        layers.forEach((layer, index, records) => {
+            layerToCreate = this._mapClassName(layer.class_name);
+            let layerNode = this.core.createNode({
+                parent: this.importedArch,
+                base: this.META[layerToCreate]
+            });
+            this.logger.debug(`Added ${layerToCreate}\
+            to ${this.core.getAttribute(this.importedArch, 'name')}`);
+
+            // Add all attributes, from the model JSON, as well as from the layers schema
+            this._addLayerAttributes(layerNode, layer);
+            this._addConfigurableNodes(layerNode, layer);
+        });
+    };
+
+    // All the attributes, which do not require a separate node to be created
+    // 1. First find the validAttributeNames for the layer
+    // 2. If the name is in layer, set it.
+    // 3. If the name is in layer.config, set it.
+    // 4. Finally, check the layers schema for remaining attributes.
+    ImportKeras.prototype._addLayerAttributes = function (layerNode, attrObj) {
+        let config = attrObj.config;
+        let validAttributeNamesForThisLayer = this.core.getValidAttributeNames(layerNode);
+        let configKeys = Object.keys(config);
+        let remainingKeys = Object.keys(attrObj)
+            .filter(value => value !== 'config');
+
+        validAttributeNamesForThisLayer.forEach((attribute) => {
+            if (remainingKeys.indexOf(this._mapArgName(attribute)) > -1) {
+                this.core.setAttribute(layerNode, attribute, this._Stringify(attrObj[this._mapArgName(attribute)]));
+                this.logger.debug(`Set ${attribute} for ${this.core.getGuid(layerNode)}` +
+                    ` to ${this.core.getAttribute(layerNode, attribute)}`);
+            } else if (configKeys.indexOf(this._mapArgName(attribute)) > -1) {
+                this.core.setAttribute(layerNode, attribute, this._Stringify(config[this._mapArgName(attribute)]));
+                this.logger.debug(`Set ${attribute} for ${this.core.getGuid(layerNode)}` +
+                    ` to ${this.core.getAttribute(layerNode, attribute)}`);
+
+            }
+        });
+        let layerName = this.core.getAttribute(layerNode, 'name');
+        this.layerInfo[layerName] = layerNode;
+
+    };
+
+    ImportKeras.prototype._addConfigurableNodes = function (layerNode, layerConfig) {
+        let allPointerNames = this.core.getValidPointerNames(layerNode);
+        let config = layerConfig.config;
+        if (allPointerNames.length > 0) {

```

Comment:
Why is this check here? Calling `forEach` on an empty array is a no-op and not an error